.github/workflows/main.yml

```diff
@@ -0,0 +1,52 @@
+name: CI
+on: 
+  pull_request:
+  push:
+    branches:
+      - main
+      - add_GHA_based_CI
+
+jobs:
+  make_sdist:
+    name: Make SDist
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - run: python -m pip install build
+
+      - run: python -m build .
+
+      - name: Upload SDist
+        uses: actions/upload-artifact@v4
+        with:
+          name: example
+          path: dist/*.tar.gz
+
+  build_wheels:
+    name: Wheel on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+        
+      - run: python -m pip install build
+
+      - run: python -m build .
+
+      - name: Upload Wheels
+        uses: actions/upload-artifact@v4
+        with:
+          name: example-wheel-${{ matrix.os }}
+          path: dist/*.whl
```

Comment:
These can be combined into a single job and then you can just have a condition `if` statement on the upload of the sdist that checks to see if the OS is `ubutnu-latest` and if so then uploads the sdist.

Reply:
Multiple jobs matches the future structure better, though. For a pure Python job you'd just have one ubuntu job that builds the wheel and SDist, and one upload. But for binaries you usually split it out rather than trying to just pick one to make the SDist on.

Reply:
So... should I combine them into one job? Or leave it as is?

Reply:
Keeping them as 2 jobs is fine.